src/Mod/TechDraw/Gui/TaskSelectLineAttributes.cpp

```diff
@@ -0,0 +1,412 @@
+/***************************************************************************
+ *   Copyright (c) 2021 edi                                                *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#ifndef _PreComp_
+#include <cmath>
+#include <BRepBndLib.hxx>
+#include <Bnd_Box.hxx>
+
+#endif // #ifndef _PreComp_
+
+#include <BRepBuilderAPI_MakeEdge.hxx>
+
+#include <QButtonGroup>
+#include <QStatusBar>
+#include <QGraphicsScene>
+
+#include <Base/Console.h>
+#include <Base/Tools.h>
+#include <Base/UnitsApi.h>
+
+#include <Gui/Application.h>
+#include <Gui/BitmapFactory.h>
+#include <Gui/Command.h>
+#include <Gui/Control.h>
+#include <Gui/Document.h>
+#include <Gui/MainWindow.h>
+#include <Gui/Selection.h>
+#include <Gui/ViewProvider.h>
+#include <Gui/WaitCursor.h>
+
+#include <Mod/TechDraw/App/DrawPage.h>
+#include <Mod/TechDraw/App/DrawUtil.h>
+#include <Mod/TechDraw/App/DrawView.h>
+#include <Mod/TechDraw/App/DrawViewPart.h>
+#include <Mod/TechDraw/App/Geometry.h>
+#include <Mod/TechDraw/App/Cosmetic.h>
+
+#include <Mod/TechDraw/Gui/ui_TaskSelectLineAttributes.h> 
+
+#include "DrawGuiStd.h"
+#include "PreferencesGui.h"
+#include "QGVPage.h"
+#include "QGIView.h"
+#include "QGIPrimPath.h"
+#include "MDIViewPage.h"
+#include "ViewProviderPage.h"
+#include "ViewProviderViewPart.h"
+#include "Rez.h"
+
+#include "TaskSelectLineAttributes.h"
+
+using namespace Gui;
+using namespace TechDraw;
+using namespace TechDrawGui;
+
+#define solid 1
```

Comment:
Is there a reason not use enums or enum classes? In modern C++ code this kind of defines should be avoided.